src/libraries/ChainlinkPath.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.17;
+
+import {BytesLib} from "./BytesLib.sol";
+import {ChainlinkOracleImpl} from "../chainlink/oracle/ChainlinkOracleImpl.sol";
+import {AggregatorV3Interface} from "chainlink/interfaces/AggregatorV3Interface.sol";
+
+/// @title Chainlink Path Library
+/// @author 0xSplits
+/// @notice Utilities for converting Chainlink Feed paths to and from bytes
+library ChainlinkPath {
+    /// -----------------------------------------------------------------------
+    /// libraries
+    /// -----------------------------------------------------------------------
+
+    using BytesLib for bytes;
+
+    /// -----------------------------------------------------------------------
+    /// constants
+    /// -----------------------------------------------------------------------
+
+    /// @dev size of one packed encoded feed
+    uint24 private constant PATH_UNIT_SIZE = 25;
+
+    /// -----------------------------------------------------------------------
+    /// functions
+    /// -----------------------------------------------------------------------
+
+    /// @notice get the number of feeds in the path
+    function len(bytes memory path) internal pure returns (uint256) {
+        return path.len(PATH_UNIT_SIZE);
+    }
+
+    /// @notice get feeds from a path (packed encoded bytes)
+    function getFeeds(bytes memory path) internal pure returns (ChainlinkOracleImpl.Feed[] memory feeds) {
+        uint256 length = len(path);
+        feeds = new ChainlinkOracleImpl.Feed[](length);
+        for (uint256 i; i < length;) {
+            feeds[i] = getFeed(path, i);
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /// @notice get a single feed from a path at a given index
+    function getFeed(bytes memory path, uint256 index) internal pure returns (ChainlinkOracleImpl.Feed memory feed) {
+        uint256 offset = index * PATH_UNIT_SIZE;
+        feed.feed = AggregatorV3Interface(path.toAddress(offset));
+        feed.staleAfter = path.toUint24(offset + 20);
+        feed.decimals = path.toUint8(offset + 23);
+        feed.mul = path.toUint8(offset + 24) == 2;
```

Comment:
why 2?

Reply:
(vs 1/0)

Reply:
so I actually had it as 1/0 but it was causing an issue with the encoding and decoding with packed mode on which is why I switched to a more explicit 2/1 approach

Reply:
huh odd.. remember any more details about the issue?

Reply:
so basically if we set mul to false it packs the encoding further to ignore those zero bits which are at the end

Reply:
huh.. that seems.. wrong? i can't imagine it just dropping the bytes bc they're zeros..